x-pack/legacy/plugins/apm/public/components/shared/MetadataTable/helper.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get, pick, isEmpty } from 'lodash';
+import { Section as SectionType } from './sections';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import { APMError } from '../../../../typings/es_schemas/ui/APMError';
+import { Span } from '../../../../typings/es_schemas/ui/Span';
+import { flattenObject, FlattenItems } from '../../../utils/flattenObject';
+
+type Item = Transaction | APMError | Span;
+
+export type MetadataItems = Array<SectionType & { data?: FlattenItems }>;
+
+export const getMetadataItems = (
+  sections: SectionType[],
+  item: Item
+): MetadataItems => {
+  return sections
+    .map(section => {
+      const sectionData: Record<string, unknown> = get(item, section.key);
+      const sectionProperties:
+        | Record<string, unknown>
+        | undefined = section.properties
+        ? pick(sectionData, section.properties)
+        : sectionData;
+      let data: FlattenItems | undefined;
+      if (!isEmpty(sectionProperties)) {
+        data = flattenObject(sectionProperties, section.key);
+      }
```

Comment:
Can't you unconditionally call `flattenObject`? It should return `[]` if `sectionProperties` is `undefined`